include/libpmemobj++/experimental/mpsc_queue.hpp

```diff
@@ -156,27 +195,126 @@ class mpsc_queue {
 		bool
 		try_produce(size_t size, Function &&f)
 		{
-			entry dram_entry;
```

Comment:
hm, can't you implememt this as `return try_produce(pmem::obj::string_view(data.get(), size))` ?